CosmicReco/inc/CosmicTrackFinderData.hh

```diff
@@ -0,0 +1,144 @@
+//Author: S Middleton 
+//Purpose: Store details of cosmic track
+#ifndef CosmicTrackFinderData_HH
+#define CosmicTrackFinderData_HH
+#include "BTrk/TrkBase/TrkErrCode.hh"
+#include "BTrk/TrkBase/TrkParticle.hh"
+#include "RecoDataProducts/inc/TrkFitDirection.hh"
+#include "RecoDataProducts/inc/StrawHitPositionCollection.hh"
+#include "RecoDataProducts/inc/StrawHitFlagCollection.hh"
+#include "RecoDataProducts/inc/StrawHitCollection.hh"
+#include "RecoDataProducts/inc/StrawHitIndex.hh"
+#include "RecoDataProducts/inc/ComboHit.hh"
+#include "RecoDataProducts/inc/StrawHit.hh"
+#include "MCDataProducts/inc/StrawDigiMC.hh"
+#include "RecoDataProducts/inc/CosmicTrackSeed.hh" //CHANGE SEED
+#include "Mu2eUtilities/inc/BuildMatrixSums.hh"
+#include "TrkReco/inc/TrkFaceData.hh"
+
+#include "Math/VectorUtil.h"
+#include "Math/Vector2D.h"
+//c++
+#include <array>
+
+using namespace ROOT::Math::VectorUtil;
+
+namespace mu2e {
+  typedef ROOT::Math::XYVectorF  XYVec;
+  // struct for weighted positions
+  class XYWVec : public XYVec {
+  public :
+    XYWVec(XYZVec pos, int face, float weight=1.0) : XYVec(pos.x(),pos.y()), _face(face), _weight(weight){}
+    float weight() const { return _weight; }
+    int   face() const { return _face; }
+
+  private :
+    int   _face;
+    float _weight; // weight for this position
+  };
+
+  class TimeCluster;
+  class CosmicTrackFinderData {
+  public:
+    
+    enum { kMaxResidIndex = 500 };
+
+    constexpr static uint16_t        kNMaxChHits = 150;
+    
+    //possible use for alignment so keep for now:
+    struct ChannelID {
+      int Station;
```

Comment:
will remove

Reply:
ahh, it turns out this is used for the ordering of the hits so is needed for the fitting routine....is it ok to leave for now. This is the same way its done in helix finder, I didnt think it would be a bad idea to mirror